### PR TITLE
320 Rounds: Replace GetCreatorForRound with GetCreator

### PR DESCRIPTION
**Pull request by algonautshant**
resolves https://app.zenhub.com/workspaces/algorand-main-workspace-5ea71fda6d554d53dda36f16/issues/algorand/go-algorand-internal/1938

### CODE REVIEW

File: ledger/acctupdates.go [hunk 1]
```diff
@@ -527,9 +527,11 @@ func (au *accountUpdates) onlineTop(rnd basics.Round, voteRnd basics.Round, n ui
 	}
 }
 
-// GetCreatorForRound returns the creator for a given asset/app index at a given round
-func (au *accountUpdates) GetCreatorForRound(rnd basics.Round, cidx basics.CreatableIndex, ctype basics.CreatableType) (creator basics.Address, ok bool, err error) {
-	return au.getCreatorForRound(rnd, cidx, ctype, true /* take the lock */)
+// GetCreator returns the creator for a given asset/app index at a given round
+func (au *accountUpdates) GetCreator(cidx basics.CreatableIndex, ctype basics.CreatableType) (creator basics.Address, ok bool, rnd basics.Round, err error) {
+	creator, ok, err = au.getCreatorForRound(au.latest(), cidx, ctype, true /* take the lock */)
+	rnd = au.latest()
```
Comment (cce) [inline ledger/acctupdates.go:533]: latest could have advanced between these two calls — could you move rnd = au.latest() up like
```suggestion
	rnd = au.latest()
	creator, ok, err = au.getCreatorForRound(rnd, cidx, ctype, true /* take the lock */)
```

File: ledger/internal/eval.go [hunk 2]
```diff
@@ -157,12 +157,14 @@ func (x *roundCowBase) getCreator(cidx basics.CreatableIndex, ctype basics.Creat
...
 	}
 
-	address, exists, err := x.l.GetCreatorForRound(x.rnd, cidx, ctype)
+	address, exists, rnd, err := x.l.GetCreator(cidx, ctype)
 	if err != nil {
 		return basics.Address{}, false, fmt.Errorf(
 			"roundCowBase.getCreator() cidx: %d ctype: %v err: %w", cidx, ctype, err)
 	}
-
+	if exists && rnd != x.rnd {
```
Comment (algonautshant) [inline ledger/internal/eval.go:165]: This condition here is true for some cases, but false when roundCowBase has the round `round-1` in `MakeDebugBalances`.

**Comment by cce** [comment]
I don't think we should be removing the ability to access rounds like this from the ledger API, after some discussion with @jannotti on wanting to support dryrun-like features or REST API lookups like https://github.com/algorand/go-algorand/pull/4018 on historical rounds that are still in deltas (and the upcoming configurable deltas parameter)

**Comment by jannotti** [comment]
> I don't think we should be removing the ability to access rounds like this from the ledger API, after some discussion with @jannotti on wanting to support dryrun-like features or REST API lookups like #4018 on historical rounds that are still in deltas (and the upcoming configurable deltas parameter)

Right - The whole reason that we're going to have MaxAccountLookback is so that these slightly old rounds can be accessed by a slightly wider REST API, or a speculative execution mechanism that needs access to a consistent round, even as it ages a little.

**Pull request closed without merging**